### ISSUE TITLE
Result contains list of unbundled files as well/

### ISSUE THREAD

**Issue by TheWix** [dowjones/gulp-bundle-assets#90]
Hey there, it would be be cool to be able to get a list of files that went into the bundle in the result file. in Dev we like to have the individual files rather than trying to debug a big bundle, and I could use the result file to print the individual scripts.

**Comment by chmontgomery** [comment]
Are you able to debug using sourcemaps? In the past that has been enough for me, at least in Chrome dev tools, because you can put break points in the original file.

That said, this enhancement wouldn't be difficult to accomplish considering we're already printing each file during ingestion. I don't have time right now but I accept PRs.

**Comment by TheWix** [comment]
Normally yes, we find however, that debugging with typescript and sourcemaps to be buggy at times so  we end up debugging the raw JS. Cool, I will look at coming up with a PR. Thanks, bud.

**Comment by chmontgomery** [comment]
@TheWix makes sense. I've also had problems in the past debugging transpiled code (babel, etc).

For the result file, I'm thinking it would make sense to _not_ have the original files by default. You could enable this functionality by adding a new [bundle option](https://github.com/dowjones/gulp-bundle-assets/blob/master/docs/API.md#bundleoptions). The actual code change could look similar to [this recent PR](https://github.com/dowjones/gulp-bundle-assets/pull/86). Basically you would end up passing down the entire src file list as a property on the streaming files. The biggest thing is that this could slow down the gulp processing if the list is huge, so again keep it as an option that is off by default (and doesn't pass the src files down).

**Comment by AndrewCraswell** [comment]
+1, my team has been needing this enhancement as well. In addition, we've found that our codebase is so large that doing the bundle takes 8-10 seconds. This is agonizing when you're waiting for a watch task do a rebuild.

**Comment by AndrewCraswell** [comment]
I've placed a $75 bounty on this issue.
https://www.bountysource.com/issues/34768377-result-contains-list-of-unbundled-files-as-well

**Comment by PlasmaPower** [comment]
I've begun work on this issue. I expect to be done by tomorrow, but things may come up.

**Comment by PlasmaPower** [comment]
I think I have a good idea of how to do this, but where should I add this into the results JSON? Every key seems to be user generated. It looks like hard coding a value might be our only option. We could also change the format of the file when the option is enabled, but I wouldn't recommend that as it'd cause more issues than a hard coded key.

**Comment by gentoo90** [comment]
Hey guys, any news on this?

**Comment by nathanwoulfe** [comment]
@PlasmaPower - is this any chance of happening? If the generated JSON file kept its existing structure but added a scriptsRaw and stylesRaw property which stored a string representing the included files, that would be amazing...

```
myBundle: {
    stylesRaw: "<link href='style1.css' media='all' rel='stylesheet' type='text/css'/><link href='style2.css' ... />"
},
```

I'm in the same boat as other commenters, where the bundling process takes 15+ seconds, so would prefer to be able to render the source files in dev, but the bundles in prod.

**Comment by PlasmaPower** [comment]
I'm coming back to this issue after a while, so I might be missing something, but I think at that time I was thinking that a user might name a bundle part `scriptsRaw`, conflicting with this.

**Comment by PlasmaPower** [comment]
Looking at this with a fresh perspective, I think it might be best to add `outputRaw`and `rawPostfix` (defaulting to `"Raw"`) options, so a user could set it to something that doesn't conflict with their existing bundle parts.

**Comment by PlasmaPower** [comment]
I'm working on this again.

**Comment by PlasmaPower** [comment]
PR created! #96

**Comment by nathanwoulfe** [comment]
Champion.

**Comment by PlasmaPower** [comment]
Would someone mind testing out my branch and see if it works for you? Since the PR doesn't seem to be getting much attention, I'd like some feedback.

**Comment by nathanwoulfe** [comment]
Off work this week, will have a look when I'm back. Sounds like exactly what I needed though, so thanks in advance.

**Comment by chmontgomery** [comment]
just published in version `2.28.0`. Let me know how it looks!
